bin/makeDocs.bash

```diff
@@ -3,118 +3,72 @@
 source colors.bash
 source lib.bash
 
-PWD="$(pwd)"
 
-# Get the path to the main repo directory.
-SCRIPT_PATH=$(readlink -f "$0")                     # Path to this script.
-SCRIPT_DIR=$(dirname "$SCRIPT_PATH")                # Path to directory containing this script.
-REPO_ROOT_DIR=$(builtin cd "$SCRIPT_DIR/.." && pwd) # REPO root directory.
-
-echo -e "${GREEN}Generating FarmData2 documentation.${NO_COLOR}."
+# Setting Env Variables
+PWD="$(pwd)" #Current Working Dir
+SCRIPT_PATH=$(readlink -f "$0") # Script Path
+SCRIPT_DIR=$(dirname "$SCRIPT_PATH") # Script Directory
+REPO_ROOT_DIR=$(builtin cd "$SCRIPT_DIR/.." && pwd) #Repository Root Directory
+
+TARGETS=()
+
+# Parsing all command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --target) TARGETS+=("$2"); shift ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
 
-echo "  Deleting old docs..."
-rm -rf docs/components 2 &> /dev/null
-mkdir docs/components
-rm -rf docs/library 2 &> /dev/null
-mkdir docs/library
-echo "  Deleted."
+INDEX_FILE="FarmData2.md" 
+INDEX_PATH="$REPO_ROOT_DIR/docs/$INDEX_FILE"
 
-echo "  Creating index file: $INDEX_FILE..."
-INDEX_FILE="FarmData2.md"
-INDEX_PATH="$REPO_ROOT_DIR/docs/$INDEX_FILE" # Absolute path.
+# Recreate the index file (Erases and rewrite into header upon regeneration)
 echo "# FarmData2 Documentation" > "$INDEX_PATH"
 echo "" >> "$INDEX_PATH"
-echo "  Created."
 
-echo "  Generating docs for all Components..."
-echo "    Adding Components section to $INDEX_FILE..."
+update_docs() {
+    local TYPE=$1 # components or library
+    local NAME=$2
+    local PATH="$REPO_ROOT_DIR/$TYPE/$NAME"
+    local DOC_PATH="docs/$TYPE/$NAME.md"
+
+    if [[ -d "$PATH" ]]; then
+        echo "    Generating docs for $NAME..."
+        # Use vue-docgen or jsdoc2md based on type
+        if [ "$TYPE" == "components" ]; then
+            npx vue-docgen "$PATH/$NAME.vue" -o "$REPO_ROOT_DIR/$DOC_PATH"
```

Comment:
For some reason I'm getting the following error here (and similarly on line 43:

<img width="819" alt="image" src="https://github.com/FarmData2/FarmData2/assets/16962482/d71cafce-920b-45cc-ba76-92b8eebf5a5d">

This results in no doc (`.md`) files being generated.

But if I run `npx` from the command line it works fine.  Are you experiencing anything similar?

Reply:
I think I've figured it out why... the error with the .md files not being generated might be related to the working directory or environment when npx is being called. Since it works fine when you run npx from the command line, it suggests that the issue is with how the environment is set up in the script. I'm going to chain the command to first `cd` from the root dir like so `(cd "$REPO_ROOT_DIR" && npx vue-docgen "$PATH/$NAME.vue" -o "$DOC_PATH")` and the same thing for jsdoc2md